webidl-napi-inl.h

```diff
@@ -435,5 +428,74 @@ inline napi_ref InstanceData::GetConstructor(const char* name) {
   return ctors[name];
 }
 
+// static
+template <typename T>
+napi_status Wrapping<T>::Create(napi_env env,
+                                napi_value js_rcv,
+                                T* cc_rcv,
+                                size_t same_obj_count) {
+  Wrapping<T>* wrapping = new Wrapping<T>;
+  wrapping->native = cc_rcv;
+  if (same_obj_count > 0)
+    wrapping->refs.resize(same_obj_count, nullptr);
+  return napi_wrap(env, js_rcv, wrapping, Destroy, nullptr, nullptr);
+}
+
+// static
+template <typename T>
+napi_status Wrapping<T>::Retrieve(napi_env env,
+                                  napi_value js_rcv,
+                                  T** cc_rcv,
+                                  int ref_idx,
```

Comment:
optional suggestion: I think I eventually figured out how it worked, but this system is pretty hard to follow. Seems like you assign a `ref_idx` to each member that has `[SameObject]`, and then store the ref for that value in `refs[ref_idx]` for that member. Could it be possible to have the `refs` members instead be actual named `napi_ref` members in a generated struct?

Current code looks reasonably correct though.

Reply:
I could go with a std::map instead.

Reply:
If the map would be keyed on a string, I think the current code is better.